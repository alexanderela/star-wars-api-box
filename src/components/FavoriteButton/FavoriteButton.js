import React from 'react';
import { NavLink } from 'react-router-dom';

import './FavoriteButton.css';
import PropTypes from 'prop-types';

const FavoriteButton = ({ favorites, handleFavoritesClick, favoritesSelected }) => (
  <div className="FavoriteButton">
    <button
      className={`cat-button favorites-button 
            ${favoritesSelected
        ? 'cat-button-active'
        : 'cat-button-inactive'}`}
      name="favorites"
      onClick={e => handleFavoritesClick(e)}
    >
      <NavLink to="/favorites" className="nav">Favorites</NavLink>
      <span className="favorites-count">
        {favorites ? favorites.length : 0}
      </span>
    </button>
  </div>
);

FavoriteButton.propTypes = {
  favorites: PropTypes.oneOfType([
    PropTypes.array,
    PropTypes.number,
  ]),
  handleFavoritesClick: PropTypes.func,
  favoritesSelected: PropTypes.bool,
};

export default FavoriteButton;
