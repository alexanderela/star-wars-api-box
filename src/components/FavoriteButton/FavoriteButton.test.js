/* eslint-disable */
import React from 'react';
import ReactDOM from 'react-dom';
import FavoriteButton from './FavoriteButton';
import { shallow, mount } from 'enzyme';

describe('FavoriteButton', () => {
	
	it('matches the snapshot', () => {
		let mockFavoriteCount = []
		let wrapper = shallow(<FavoriteButton favoritesCount={[mockFavoriteCount]}/>);
		expect(wrapper).toMatchSnapshot();
	})

  it('calls the handleFavoritesClick function on click', () => {
    let mockFavsClick = jest.fn()
    let mockFavoritesSelected = true
    let mockFavs = [{
          name: "Dina Caraballo",
          id: "Dina Caraballo",
          isFavorite: true, 
          type:"people",
          properties: [
            {header: 'Homeworld: ', text: "Earth"},
            {header: 'Species: ', text: "human"},
            {header: 'Population: ', text: "7,000,000,000"},
            {header: 'Language: ', text: "Galactic Basic"}
          ] 
        },
        {
          name: "Alex Ela",
          id: "Alex Ela",
          isFavorite: true, 
          type:"people",
          properties: [
            {header: 'Homeworld: ', text: "Earth"},
            {header: 'Species: ', text: "human"},
            {header: 'Population: ', text: "7,000,000,000"},
            {header: 'Language: ', text: "Galactic Basic"}
          ] 
        }, 
        {
          "name": "Luke Skywalker",
          "id": "Luke Skywalker",
          "isFavorite": true, 
          "type":"people",
          "properties": [
            {"header": 'Homeworld: ', "text": "Tatooine"},
            {"header": 'Species: ', "text": "human"},
            {"header": 'Population: ', "text": "200,000"},
            {"header": 'Language: ', "text": "Galactic Basic"}
            ] 
          }]
    let wrapper = shallow(
      <FavoriteButton 
        handleFavoritesClick={mockFavsClick}
        favoritesSelected={mockFavoritesSelected}
        favorites={mockFavs}
      />);
    wrapper.find('button').simulate('click')
    expect(wrapper.find('button').hasClass('fav-btn-inactive')).toBe(false)

  })
})
