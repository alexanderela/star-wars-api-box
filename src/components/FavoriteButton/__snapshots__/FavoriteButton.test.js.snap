// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FavoriteButton matches the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <FavoriteButton
    favoritesCount={
      Array [
        Array [],
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <button
        className="cat-button favorites-button 
            cat-button-inactive"
        name="favorites"
        onClick={[Function]}
      >
        <NavLink
          activeClassName="active"
          aria-current="page"
          className="nav"
          to="/favorites"
        >
          Favorites
        </NavLink>
        <span
          className="favorites-count"
        >
          0
        </span>
      </button>,
      "className": "FavoriteButton",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <NavLink
            activeClassName="active"
            aria-current="page"
            className="nav"
            to="/favorites"
          >
            Favorites
          </NavLink>,
          <span
            className="favorites-count"
          >
            0
          </span>,
        ],
        "className": "cat-button favorites-button 
            cat-button-inactive",
        "name": "favorites",
        "onClick": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "activeClassName": "active",
            "aria-current": "page",
            "children": "Favorites",
            "className": "nav",
            "to": "/favorites",
          },
          "ref": null,
          "rendered": "Favorites",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": 0,
            "className": "favorites-count",
          },
          "ref": null,
          "rendered": 0,
          "type": "span",
        },
      ],
      "type": "button",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <button
          className="cat-button favorites-button 
            cat-button-inactive"
          name="favorites"
          onClick={[Function]}
        >
          <NavLink
            activeClassName="active"
            aria-current="page"
            className="nav"
            to="/favorites"
          >
            Favorites
          </NavLink>
          <span
            className="favorites-count"
          >
            0
          </span>
        </button>,
        "className": "FavoriteButton",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <NavLink
              activeClassName="active"
              aria-current="page"
              className="nav"
              to="/favorites"
            >
              Favorites
            </NavLink>,
            <span
              className="favorites-count"
            >
              0
            </span>,
          ],
          "className": "cat-button favorites-button 
            cat-button-inactive",
          "name": "favorites",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "activeClassName": "active",
              "aria-current": "page",
              "children": "Favorites",
              "className": "nav",
              "to": "/favorites",
            },
            "ref": null,
            "rendered": "Favorites",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": 0,
              "className": "favorites-count",
            },
            "ref": null,
            "rendered": 0,
            "type": "span",
          },
        ],
        "type": "button",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
